### PR TITLE
"5 for $50" Sale Calculator

### PR DESCRIPTION
**Pull request by iloveitaly**
This is a _really rough_ version of a "5 for $50" sale calculator. Wanted to gauge interest and see if this is something that would be included in core or is better fitted for a extension.

If this is something that could be included in core I'll clean it up and write some tests.

### CODE REVIEW

**Comment by BenMorganIO** [comment]
I currently require this promotion for a client. I'll be willing to put in some time to get this PR mergeable.

**Comment by iloveitaly** [comment]
> I currently require this promotion for a client. I'll be willing to put in some time to get this PR mergeable.

Go for it. Could you write the specs and submit a PR to my forked repo?

**Comment by BenMorganIO** [comment]
@iloveitaly I'll try to get the test coverage increased. I'll make the PRs against your branch.

**Comment by BenMorganIO** [comment]
@iloveitaly I might actually cherry-pick your commits to a new branch since this is PR is so old.

File: core/app/models/spree/calculator/n_for_x.rb [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      end
+
+      def matching_variants
+        @matching_variants ||= if compute_on_promotion?
+          self.calculable.promotion.rules.select { |r| r.respond_to?(:variants) }.map(&:variants).flatten
+        end
+      end
+
+      def compute_on_promotion?
+        return true
```
Comment (BenMorganIO) [inline core/app/models/spree/calculator/n_for_x.rb]: @iloveitaly Why are you returning true indefinitely, but there's code below it?

Comment (BenMorganIO) [inline core/app/models/spree/calculator/n_for_x.rb]: Is there a missing if statement?

File: core/app/models/spree/calculator/n_for_x.rb [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    end
+
+    private
+
+      def applicable_variants(object = nil)
+        return @applicable_variants if !@applicable_variants.nil?
+
+        @applicable_variants = []
+
+        matching_line_items(object).each do |l|
```
Comment (BenMorganIO) [inline core/app/models/spree/calculator/n_for_x.rb]: Does `l` refer to a `line_item`?

**Comment by bbonislawski** [comment]
This PR is stale and I'm closing it for know. If anyone wants to update it with specs, please let me know and I'll reopen this PR.

**Pull request closed without merging**